pkg/provisioning/parse.go

```diff
@@ -28,6 +28,7 @@ import (
 type ProcessorConfig struct {
 	Type     string            `yaml:"type"`
 	Settings map[string]string `yaml:"settings"`
+	Workers  int               `yaml:"workers"`
```

Comment:
if no workers are defined, do we default to 1 worker? also, we should add a validation for the workers to be a positive number, in `provisioning/validate.go `

Reply:
Yes, we default to 1 worker, but it's a bit further down the line ([here](https://github.com/ConduitIO/conduit/blob/aa4b1587fc7cf4592166f74f132d9b9831a55ed1/pkg/pipeline/lifecycle.go#L268)). I can add a default and validation here, makes sense. Same thing in the API endpoint probably.

Reply:
I added validations for the workers variable in the pipeline config file (`provisioning/validate.go`) and in `processor.Service` (which will bubble up to the API). Also did some cleaning up in tests (moved tests to `matryer/is`, usage of temporary folder in runtime test).